src/matching/linear-programming.jl

```diff
@@ -0,0 +1,111 @@
+"""
+An optimal matching.
+
+weight: total weight of the matching
+
+inmatch: `inmatch[e]=true` if edge `e` belongs to the matching.
+
+π:       `π[i]=j` if vertex `i` is matched to vertex `j`.
+         `π[i]=-1` for unmatched vertices.
+"""
+type MatchingResult
+    weight::Float64
+    inmatch::Dict{Edge,Bool}
+    π::Vector{Int}
+end
+
+"""
+As `maximum_weight_maximal_matching`, with the difference that the edges `e` with
+`w[e] < cutoff` will not be considered for the matching.
+"""
+function maximum_weight_maximal_matching{T<:Number}(g::Graph, w::Dict{Edge,T}, cutoff)
+    wnew = Dict{Edge,T}()
+    for (e,x) in w
+        if x >= cutoff
+            wnew[e] = x
+        end
+    end
+
+    return maximum_weight_maximal_matching(g, wnew)
+end
+
+"""
+Given a bipartite graph `g` and an edgemap `w` containing weights associated to edges,
+returns a matching with the maximum total weight among the ones containing the
+greatest number of edges.
+Edges in `g` not present in `w` will not be considered for the matching.
+The algorithm relies on a linear relaxation on of the matching problem, which is
+guaranteed to have integer solution on bipartite graps.
+The pakage JuMP.jl and one of its supported solvers is required.
+"""
+function maximum_weight_maximal_matching{T<:Number}(g::Graph, w::Dict{Edge,T})
+# TODO support for graphs with zero degree nodes
+# TODO apply separately on each connected component
+    isdefined(:JuMP) || error("JuMP not defined")
+    bpmap = bipartite_map(g)
+    length(bpmap) != nv(g) && error("Graph is not bipartite")
+    v1 = findin(bpmap, 1)
+    v2 = findin(bpmap, 2)
+    if length(v1) > length(v2)
+        v1, v2 = v2, v1
+    end
+    nedg = 0
+    edgemap = Dict{Edge,Int}([e => nedg+=1 for (e,w) in w])
+    for (e,idx) in edgemap
+        edgemap[reverse(e)] = idx
+    end
```

Comment:
symetrizing a `Dict{Edge,T}` is a reusable routine that we will probably need in the future. Go ahead and make it a function with some tests.